packages/permissions/src/utils/has-permissions.ts

```diff
@@ -141,3 +172,71 @@ export const getInvalidPermissions = (
     isNil(actualPermissions[toCanCase(demandedPermission)])
   );
 };
+
+const hasDemandedDataFenceForStore = (
+  resourceToApplyDataFence: TResourceToApplyDataFence,
+  options: {
+    actualDataFence: TDataFenceByPermissionName;
+    actualDataFenceName: string;
+    demandedDataFence: TDemandedDataFence;
+  }
+): boolean => {
+  const hasDemandedPermission = hasPermission(options.demandedDataFence.name, {
+    [options.actualDataFenceName]: true,
+  });
+  if (!hasDemandedPermission) return false;
+  switch (options.demandedDataFence.group) {
+    case 'orders': {
+      return options.actualDataFence.values.includes(
+        resourceToApplyDataFence.storeRef &&
+          resourceToApplyDataFence.storeRef.key
+          ? resourceToApplyDataFence.storeRef.key
+          : ''
+      );
+    }
+    default:
+      return false;
+  }
+};
+
+export const createHasAppliedDataFence = (
+  options: TOptionsForAppliedDataFence
+) => (resourceToApplyDataFence: TResourceToApplyDataFence): boolean => {
+  // First check that the demanded dataFence is enforced on `projectPermissions`
+  const hasDemandedProjectPermission = hasPermission(
+    options.demandedDataFence.name,
+    options.actualPermissions
+  );
+  // Given that dataFence structure on `applicationContext`, we get the value by a path
+  // e.g given dataFence with { store: { orders: { canManageOrders: { values: } } } }
+  // we read the dataFence by the [type] and [group]
+  // dataFence[type][group] = dataFence.store.group
+  // with value = there is a dataFence to apply, overrules `hasDemandedProjectPermissions`
+  // without value = there is no dataFence to apply, overruled by `hasDemandedProjectPermissions`
+  const actualDataFenceByResourceGroup = get(
+    options.actualDataFences,
+    `${options.demandedDataFence.type}.${options.demandedDataFence.group}`
+  );
+  // { canManageOrders: { values: [] }}
+  if (actualDataFenceByResourceGroup) {
+    const hasDemandedDataFence = Object.entries(
+      actualDataFenceByResourceGroup
+    ).every(([actualDataFenceName, actualDataFence]): boolean => {
+      switch (options.demandedDataFence.type) {
```

Comment:
what happens in here could be determined by the client.
that is... the implementation of `hasDemandedDataFenceForStore` can move to MC-FE if we are not happy to have it on app-kit

Reply:
I think it should be in app-kit as custom apps might want to use it too.